If there are more than 174 organizations per user, the synchronization no longer works. Limit 174 organizations per user?

<!--
    # ###
    NOTE: Please update to the latest version of vaultwarden before reporting an issue!
    This saves you and us a lot of time and troubleshooting.
    See:
    * https://github.com/dani-garcia/vaultwarden/issues/1180
    * https://github.com/dani-garcia/vaultwarden/wiki/Updating-the-vaultwarden-image
    # ###
-->

<!--
Please fill out the following template to make solving your problem easier and faster for us.
This is only a guideline. If you think that parts are unnecessary for your issue, feel free to remove them.

Remember to hide/redact personal or confidential information,
such as passwords, IP addresses, and DNS names as appropriate.
-->

### Subject of the issue

Currently I have the following problem: A user has 174 organizations. If this user has created another organization, synchronization will no longer work. All 175 organizations no longer appear in the web browser. When I try to synchronize in the Bitwarden APP, the error message appears: “Failed to synchronize”.

If I delete an organization using the admin token, everything works again without any problems.

Vaultwarden runs in a Docker container within ProxMox on the current version 2023.10.0 with the MySQL DB. In the OPNsense it is configured with Nginx.

Do you have a solution for the problem?


### Deployment environment

<!--
    =========================================================================================
    Preferably, use the `Generate Support String` button on the admin page's Diagnostics tab.
    That will auto-generate most of the info requested in this section.
    =========================================================================================
-->

<!-- The version number, obtained from the logs (at startup) or the admin diagnostics page -->
2023.10.0 
<!-- This is NOT the version number shown on the web vault, which is versioned separately from vaultwarden -->
2023.10.0 
<!-- Remember to check if your issue exists on the latest version first! -->
* vaultwarden version: 2023.10.0 

<!-- How the server was installed: Docker image, OS package, built from source, etc. -->
* Install method: Docker image

* Clients used: <!-- web vault, desktop, Android, iOS, etc. (if applicable) -->
* Google Chrome
* Firefox
* Windows-App
* iOS APP

* Reverse proxy and version: <!-- if applicable -->
* Nginx on OPNsnese

* MySQL/MariaDB or PostgreSQL version: <!-- if applicable -->
* MySQL

* Other relevant details:

### Steps to reproduce
<!-- Tell us how to reproduce this issue. What parameters did you set (differently from the defaults)
and how did you start vaultwarden? -->


Everything is set to default. If you create the 175 organization for a user, the error occurs.

### Expected behaviour
<!-- Tell us what you expected to happen -->

### Actual behaviour
<!-- Tell us what actually happened -->

### Troubleshooting data
<!-- Share any log files, screenshots, or other relevant troubleshooting data -->

Hi BlackDex

Here are the logs

[2023-12-11 12:48:52.735][request][INFO] GET /api/config
[2023-12-11 12:48:52.735][response][INFO] (config) GET /api/config => 200 OK
[2023-12-11 12:48:54.492][request][INFO] GET /api/devices/knowndevice
[2023-12-11 12:48:54.492][response][INFO] (get_known_device) GET /api/devices/knowndevice => 200 OK
[2023-12-11 12:48:55.460][request][INFO] POST /api/organizations
[2023-12-11 12:48:55.462][response][INFO] (create_organization) POST /api/organizations => 200 OK
[2023-12-11 12:48:55.489][request][INFO] POST /identity/connect/token
[2023-12-11 12:48:55.490][response][INFO] (login) POST /identity/connect/token => 200 OK
[2023-12-11 12:48:55.650][rocket::data::data::_][ERROR] Failed to read into peek buffer: Custom { kind: Other, error: hyper::Error(Body, Custom { kind: UnexpectedEof, error: IncompleteBody }) }.
[2023-12-11 12:48:55.650][request][INFO] POST /identity/connect/token
[2023-12-11 12:48:55.650][vaultwarden::api::identity::_][WARN] Data guard `Form < ConnectData >` failed: Errors([Error { name: Some("grant_type"), value: None, kind: Missing, entity: Field }]).
[2023-12-11 12:48:55.650][response][INFO] (login) POST /identity/connect/token => 422 Unprocessable Entity
[2023-12-11 12:49:01.248][request][INFO] POST /identity/accounts/prelogin
[2023-12-11 12:49:01.248][response][INFO] (prelogin) POST /identity/accounts/prelogin => 200 OK
[2023-12-11 12:49:01.502][request][INFO] POST /identity/connect/token
[2023-12-11 12:49:01.542][vaultwarden::api::identity][INFO] User support@medheads.de logged in successfully. IP: x.x.x.x
[2023-12-11 12:49:01.542][response][INFO] (login) POST /identity/connect/token => 200 OK
[2023-12-11 12:49:16.358][vaultwarden::api::notifications][INFO] Closing WS connection from 46.114.2.149
[2023-12-11 12:49:16.358][rocket::server][ERROR] Upgraded websocket I/O handler failed: WebSocket protocol error: Sending after closing is not allowed
[2023-12-11 12:50:59.771][request][INFO] GET /api/devices/knowndevice
[2023-12-11 12:50:59.771][response][INFO] (get_known_device) GET /api/devices/knowndevice => 200 OK
[2023-12-11 12:51:06.313][request][INFO] POST /identity/accounts/prelogin
[2023-12-11 12:51:06.314][response][INFO] (prelogin) POST /identity/accounts/prelogin => 200 OK
[2023-12-11 12:51:06.541][request][INFO] POST /identity/connect/token
[2023-12-11 12:51:06.581][vaultwarden::api::identity][INFO] User support@medheads.de logged in successfully. IP: x.x.x.x
[2023-12-11 12:51:06.581][response][INFO] (login) POST /identity/connect/token => 200 OK
[2023-12-11 12:51:06.731][request][INFO] POST /identity/connect/token
[2023-12-11 12:51:06.732][response][INFO] (login) POST /identity/connect/token => 200 OK

Does this also happen at the browser/web-vault? -> Yes, that happens everywhere. Web browser, iOS APP, Windows APP

Developer Console  -> Unfortunately I can't say that because I don't know how to use it

--- a/src/api/identity.rs
+++ b/src/api/identity.rs
@@ -103,8 +103,13 @@ async fn _refresh_login(data: ConnectData, conn: &mut DbConn) -> JsonResult {
 
     // Common
     let user = User::find_by_uuid(&device.user_uuid, conn).await.unwrap();
-    let orgs = UserOrganization::find_confirmed_by_user(&user.uuid, conn).await;
-    let (access_token, expires_in) = device.refresh_tokens(&user, orgs, scope_vec);
+    // ---
+    // Disabled this variable, it was used to generate the JWT
+    // Because this might get used in the future, and is add by the Bitwarden Server, lets keep it, but then commented out
+    // See: https://github.com/dani-garcia/vaultwarden/issues/4156
+    // ---
+    // let orgs = UserOrganization::find_confirmed_by_user(&user.uuid, conn).await;
+    let (access_token, expires_in) = device.refresh_tokens(&user, scope_vec);
     device.save(conn).await?;
 
     let result = json!({
@@ -260,8 +265,13 @@ async fn _password_login(
     }
 
     // Common
-    let orgs = UserOrganization::find_confirmed_by_user(&user.uuid, conn).await;
-    let (access_token, expires_in) = device.refresh_tokens(&user, orgs, scope_vec);
+    // ---
+    // Disabled this variable, it was used to generate the JWT
+    // Because this might get used in the future, and is add by the Bitwarden Server, lets keep it, but then commented out
+    // See: https://github.com/dani-garcia/vaultwarden/issues/4156
+    // ---
+    // let orgs = UserOrganization::find_confirmed_by_user(&user.uuid, conn).await;
+    let (access_token, expires_in) = device.refresh_tokens(&user, scope_vec);
     device.save(conn).await?;
 
     let mut result = json!({
@@ -374,8 +384,13 @@ async fn _user_api_key_login(
 
     // Common
     let scope_vec = vec!["api".into()];
-    let orgs = UserOrganization::find_confirmed_by_user(&user.uuid, conn).await;
-    let (access_token, expires_in) = device.refresh_tokens(&user, orgs, scope_vec);
+    // ---
+    // Disabled this variable, it was used to generate the JWT
+    // Because this might get used in the future, and is add by the Bitwarden Server, lets keep it, but then commented out
+    // See: https://github.com/dani-garcia/vaultwarden/issues/4156
+    // ---
+    // let orgs = UserOrganization::find_confirmed_by_user(&user.uuid, conn).await;
+    let (access_token, expires_in) = device.refresh_tokens(&user, scope_vec);
     device.save(conn).await?;
 
     info!("User {} logged in successfully via API key. IP: {}", user.email, ip.ip);

--- a/src/auth.rs
+++ b/src/auth.rs
@@ -119,10 +119,16 @@ pub struct LoginJwtClaims {
     pub email: String,
     pub email_verified: bool,
 
-    pub orgowner: Vec<String>,
-    pub orgadmin: Vec<String>,
-    pub orguser: Vec<String>,
-    pub orgmanager: Vec<String>,
+    // ---
+    // Disabled these keys to be added to the JWT since they could cause the JWT to get too large
+    // Also These key/value pairs are not used anywhere by either Vaultwarden or Bitwarden Clients
+    // Because these might get used in the future, and they are added by the Bitwarden Server, lets keep it, but then commented out
+    // See: https://github.com/dani-garcia/vaultwarden/issues/4156
+    // ---
+    // pub orgowner: Vec<String>,
+    // pub orgadmin: Vec<String>,
+    // pub orguser: Vec<String>,
+    // pub orgmanager: Vec<String>,
 
     // user security_stamp
     pub sstamp: String,

--- a/src/db/models/device.rs
+++ b/src/db/models/device.rs
@@ -59,12 +59,7 @@ impl Device {
         self.twofactor_remember = None;
     }
 
-    pub fn refresh_tokens(
-        &mut self,
-        user: &super::User,
-        orgs: Vec<super::UserOrganization>,
-        scope: Vec<String>,
-    ) -> (String, i64) {
+    pub fn refresh_tokens(&mut self, user: &super::User, scope: Vec<String>) -> (String, i64) {
         // If there is no refresh token, we create one
         if self.refresh_token.is_empty() {
             use data_encoding::BASE64URL;
@@ -75,10 +70,17 @@ impl Device {
         let time_now = Utc::now().naive_utc();
         self.updated_at = time_now;
 
-        let orgowner: Vec<_> = orgs.iter().filter(|o| o.atype == 0).map(|o| o.org_uuid.clone()).collect();
-        let orgadmin: Vec<_> = orgs.iter().filter(|o| o.atype == 1).map(|o| o.org_uuid.clone()).collect();
-        let orguser: Vec<_> = orgs.iter().filter(|o| o.atype == 2).map(|o| o.org_uuid.clone()).collect();
-        let orgmanager: Vec<_> = orgs.iter().filter(|o| o.atype == 3).map(|o| o.org_uuid.clone()).collect();
+        // ---
+        // Disabled these keys to be added to the JWT since they could cause the JWT to get too large
+        // Also These key/value pairs are not used anywhere by either Vaultwarden or Bitwarden Clients
+        // Because these might get used in the future, and they are added by the Bitwarden Server, lets keep it, but then commented out
+        // ---
+        // fn arg: orgs: Vec<super::UserOrganization>,
+        // ---
+        // let orgowner: Vec<_> = orgs.iter().filter(|o| o.atype == 0).map(|o| o.org_uuid.clone()).collect();
+        // let orgadmin: Vec<_> = orgs.iter().filter(|o| o.atype == 1).map(|o| o.org_uuid.clone()).collect();
+        // let orguser: Vec<_> = orgs.iter().filter(|o| o.atype == 2).map(|o| o.org_uuid.clone()).collect();
+        // let orgmanager: Vec<_> = orgs.iter().filter(|o| o.atype == 3).map(|o| o.org_uuid.clone()).collect();
 
         // Create the JWT claims struct, to send to the client
         use crate::auth::{encode_jwt, LoginJwtClaims, DEFAULT_VALIDITY, JWT_LOGIN_ISSUER};
@@ -93,11 +95,16 @@ impl Device {
             email: user.email.clone(),
             email_verified: !CONFIG.mail_enabled() || user.verified_at.is_some(),
 
-            orgowner,
-            orgadmin,
-            orguser,
-            orgmanager,
-
+            // ---
+            // Disabled these keys to be added to the JWT since they could cause the JWT to get too large
+            // Also These key/value pairs are not used anywhere by either Vaultwarden or Bitwarden Clients
+            // Because these might get used in the future, and they are added by the Bitwarden Server, lets keep it, but then commented out
+            // See: https://github.com/dani-garcia/vaultwarden/issues/4156
+            // ---
+            // orgowner,
+            // orgadmin,
+            // orguser,
+            // orgmanager,
             sstamp: user.security_stamp.clone(),
             device: self.uuid.clone(),
             scope,

